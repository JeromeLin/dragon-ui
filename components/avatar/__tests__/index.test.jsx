import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Avatar from '../index';

describe('Avatar', () => {
  it('renders Avatars of different size correctly', () => {
    const wrapper = render(
      <div>
        <Avatar size="xl">XL</Avatar>
        <Avatar size="lg">LG</Avatar>
        <Avatar>MD</Avatar>
        <Avatar size="sm">SM</Avatar>
        <Avatar size="xs">XS</Avatar>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different shape correctly', () => {
    const wrapper = shallow(
      <div>
        <Avatar />
        <Avatar shape="circle" />
        <Avatar shape="square" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different type correctly', () => {
    const wrapper = render(
      <div>
        <Avatar size="sm" style={{ fontSize: '12px' }}>USER</Avatar>
        <Avatar size="sm" style={{ color: 'red', backgroundColor: 'rgb(249, 232, 8)' }}>USER</Avatar>
        <Avatar
          size="sm"
          src="site/images/avatar/avatar.png"
          alt="my avatar"
        />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars correctly when set src, icon and string', () => {
    const wrapper = render(
      <div>
        <Avatar
          size="sm"
          src="site/images/avatar/avatar.png"
          alt="my avatar"
        />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars correctly that Automatically adjust the character size', () => {
    const wrapper = mount(
      <Avatar size="lg">Zhong</Avatar>,
    );
    expect(wrapper.state().childrenScale).toBe(1);


    wrapper.setProps({ children: null });
    expect(wrapper.state().childrenScale).toBe(1);

    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', {
      get() {
        if (this.className === 'zw-avatar--string') {
          return 160;
        }
        return 48;
      },
    });
    wrapper.setProps({ children: 'Component' });
    expect(wrapper.state().childrenScale).toBe(0.25);

    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', {
      get() {
        if (this.className === 'zw-avatar--string') {
          return 50;
        }
        return 108;
      },
    });
    wrapper.setProps({ children: 'ZA' });
    expect(wrapper.state().childrenScale).toBe(1);
  });
});
