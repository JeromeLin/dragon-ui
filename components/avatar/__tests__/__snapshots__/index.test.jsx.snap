// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders Avatars correctly when set src, icon and string 1`] = `
<div>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--sm"
  >
    <img
      alt="my avatar"
      class="zw-avatar--image"
      src="site/images/avatar/avatar.png"
    />
  </span>
</div>
`;

exports[`Avatar renders Avatars of different shape correctly 1`] = `
<div>
  <Avatar
    prefixCls="zw-avatar"
    shape="circle"
    size="md"
  />
  <Avatar
    prefixCls="zw-avatar"
    shape="circle"
    size="md"
  />
  <Avatar
    prefixCls="zw-avatar"
    shape="square"
    size="md"
  />
</div>
`;

exports[`Avatar renders Avatars of different size correctly 1`] = `
<div>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--xl"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      XL
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--lg"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      LG
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--md"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      MD
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--sm"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      SM
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--xs"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      XS
    </span>
  </span>
</div>
`;

exports[`Avatar renders Avatars of different type correctly 1`] = `
<div>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--sm"
    style="font-size:12px"
  >
    <span
      class="zw-avatar--string"
    >
      USER
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--sm"
    style="color:red;background-color:rgb(249, 232, 8)"
  >
    <span
      class="zw-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      USER
    </span>
  </span>
  <span
    class="zw-avatar zw-avatar--circle zw-avatar--sm"
  >
    <img
      alt="my avatar"
      class="zw-avatar--image"
      src="site/images/avatar/avatar.png"
    />
  </span>
</div>
`;
